pkg/operator/ceph/cluster/watcher.go

```diff
@@ -262,6 +262,16 @@ func listRBDPV(listPVs *corev1.PersistentVolumeList, cluster *cephv1.CephCluster
 	var listRbdPV []corev1.PersistentVolume
 
 	for _, pv := range listPVs.Items {
+		// Skip if pv is not provisioned by CSI
+		if pv.Spec.CSI == nil {
+			logger.Warningf("pv %q is not provisioned by CSI", pv.Name)
+			continue
+		}
+		if pv.Spec.CSI.VolumeAttributes["pool"] == "" || pv.Spec.CSI.VolumeAttributes["imageName"] == "true" {
+			logger.Warningf("skipping, static pv %q", pv.Name)
+			continue
+		}
```

Comment:
i think the earlier check was also better, can you do like below

```suggestion
for _, pv := range listPVs.Items {
		// Skip if pv is not provisioned by CSI
	if pv.Spec.CSI == nil {
	       logger.Warningf("pv %q is not provisioned by CSI", pv.Name)
		continue
	}
if pv.Spec.CSI.Driver == fmt.Sprintf("%s.rbd.csi.ceph.com", cluster.Namespace) {
     if pvSupportsMultiNodeAccess(pv.Spec.AccessModes) {
		continue
}
if  `pv.Spec.CSI.VolumeAttributes["staticVolume"] == "true"` || pv.Spec.CSI.VolumeAttributes["pool"] == "" || pv.Spec.CSI.VolumeAttributes["imageName"] == "" {
            logger.Warningf("skipping, static pv %q", pv.Name)
			continue
}
for _, rbdVolume := range rbdVolumesInUse {
				if pv.Spec.CSI.VolumeHandle == rbdVolume {
					listRbdPV = append(listRbdPV, pv)
				}
			}
		}
return listRbdPV
}
` to line 276 where we have `pv.Spec.CSI.VolumeHandle == rbdVolume` check